src/Microsoft.OData.Edm/Validation/ValidationRules.cs

```diff
@@ -2383,54 +2383,55 @@ public static class ValidationRules
                     var target = annotation.Target;
                     bool foundTarget = false;
 
-                    IEdmEntityContainer entityContainer = target as IEdmEntityContainer;
-                    if (entityContainer != null)
+                    if (target is IEdmEntityContainer entityContainer)

```

Comment:
@marabooy After the neat approach you adopted, you could go further and cure the not-so-good-looking nesting by rewriting the if-else-if's as follows:
```csharp
if (target is IEdmEntitySet entitySet)
{
    // yada yada
}
else if (target is IEdmSingleton singleton)
{
    // yada yada
}
else if (target is IEdmSchemaType schemaType)
{
    // yada yada
}
else if (target is IEdmTerm term)
{
    // yada yada
}
// etcetra etcetra
```

Reply:
resolved